---
title: "Seurat_Spongilla_RNASeq_analysis"
output: html_document
date: "2026-01-15"
---

```{r}

## Libraries
library(Seurat)  #ver 2.3.4
library(dplyr)  #ver 0.8.1
library(ggplot2)
library(tidyverse)
library(viridis)
library(RColorBrewer)

#ENASeq analysis code modified from https://git.embl.org/musser/profiling-cellular-diversity-in-sponges-informs-animal-cell-type-and-nervous-system-evolution/-/tree/master/single_cell_analysis/data  

#import 10x matrix and sample metadata
sl_count_matrix <- read.delim(file = "/GSE134912_spongilla_10x_count_matrix.txt") #obtained from https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE134912  
sampleID_metadata <- read.delim(file = "/spongilla_cell_metadata.tsv", stringsAsFactors = F) #obtained from https://git.embl.org/musser/profiling-cellular-diversity-in-sponges-informs-animal-cell-type-and-nervous-system-evolution/-/tree/master/single_cell_analysis/data 

#initialize seurat object
sl <- CreateSeuratObject(counts= sl_count_matrix, project = "spongilla_sc")
sl <- AddMetaData(object = sl, metadata = sampleID_metadata)

Idents(sl) <- "cell_type"  # Set cell type as the active identity

#normalize and scal data
sl <- NormalizeData(sl)
sl <- FindVariableFeatures(sl)

#plotting lysosomal genes log2fold>1
DE_genes_lysosomal_sclerocyte <- read.csv("/Users/maitripaul/Dropbox/Tresguerres_lab/sponge_project/Supplementary_files/DE_sclerocyte_genes_logfold1_lysosomal_S_lacustris.csv")

genes_of_interest <- DE_genes_lysosomal_sclerocyte$gene_id

#green_purple_palette <- brewer.pal(11, "PiYG")


# Create the dot plot for the filtered Seurat object and selected genes
lysosomal_dot_plot <- DotPlot(sl_filtered, features = rev(genes_of_interest), group.by = "cell_type") +
   scale_x_discrete(labels = function(x) {
    # Match original row names to their new names
    sapply(x, function(g) {
      if (g %in% DE_genes_lysosomal_sclerocyte$gene_id) {
       DE_genes_lysosomal_sclerocyte$gene_name[DE_genes_lysosomal_sclerocyte$gene_id == g]
      } else {
        g  # Keep the original if no match
      }
    })
  })+
 scale_color_gradientn(colors = c("#CCFF00", "#999999", "#FF66FF"))+
#scale_color_gradientn(colors = c("#336600", "white", "#CC0066"))+
  ggplot2::theme_classic() +
  theme(axis.text.x = element_text(angle = 90, hjust = 1)) + coord_flip()



#plotting VHA subunits
#some Musser cell types did not have assignment and were just numbers, excluded those for interpretation
cell_types_with_letters <- grep("[a-zA-Z]", unique(sl$cell_type), value = TRUE)

# Subset the Seurat object to only include these cell types
sl_filtered <- subset(sl, cells = WhichCells(sl, expression = cell_type %in% cell_types_with_letters))

# Reorder the cell types in alphabetical order


VHA_subunit_genes <- read.csv("data/VHA_subunits_in_S_lacustris.csv")
VHA_subunit_genes <- VHA_subunit_genes %>% filter(if_all(everything(), ~ !is.na(.)))

genes_of_interest <- VHA_subunit_genes$Gene.from.transcriptome

green_purple_palette <- brewer.pal(11, "PiYG")


# Create the dot plot for the filtered Seurat object and selected genes
VHA_subunit_genes_dot_plot <- DotPlot(sl_filtered, features = rev(genes_of_interest), group.by = "cell_type") +
  #scale_y_discrete(limits = sort(unique(sl_filtered$cell_type))) + # Reorder the y-axis explicitly
  scale_x_discrete(labels = function(x) {
    # Match original row names to their new names
    sapply(x, function(g) {
      if (g %in% VHA_subunit_genes$Gene.from.transcriptome) {
       VHA_subunit_genes$VHA.subunit[VHA_subunit_genes$Gene.from.transcriptome == g]
      } else {
        g  # Keep the original if no match
      }
    })
  })+
  scale_color_gradientn(colors = c("#CCFF00", "#999999", "#FF66FF"))+
  
  ggplot2::theme_classic() +
  theme(axis.text.x = element_text(angle = 90, hjust = 1)) + coord_flip()
 



#plotting Lsi2, OCA2, NBCSA
Lsi2_genes <- read.csv("data/Supplementary_files/Lsi2_OCA2_NBCSA_in_S_lacustris.csv")


genes_of_interest_list <- Lsi2_genes$spongilla_top_hit

# Create the dot plot for the filtered Seurat object and selected genes
Lsi2_dot_plot <- DotPlot(sl_filtered, features = rev(genes_of_interest_list), group.by = "cell_type") +
  #scale_y_discrete(limits = sort(unique(sl_filtered$cell_type))) + # Reorder the y-axis explicitly
  scale_x_discrete(labels = function(x) {
    # Match original row names to their new names
    sapply(x, function(g) {
      if (g %in% Lsi2_genes$spongilla_top_hit) {
       Lsi2_genes$spongilla_gene_abrv[Lsi2_genes$spongilla_top_hit == g]
      } else {
        g  # Keep the original if no match
      }
    })
  })+
    #for white bg
#scale_color_gradientn(colors = c("#336600", "white", "#CC0066"))+
  scale_color_gradientn(colors = c("#CCFF00", "#999999", "#FF66FF"))+
  ggplot2::theme_classic() +
  theme(axis.text.x = element_text(angle = 90, hjust = 1)) + coord_flip()




```